skopt/utils.py

```diff
@@ -200,3 +210,72 @@ def func(x):
             best_fun = r.fun
 
     return [v for v in best_x], best_fun
+
+
+def has_gradients(estimator, space):
+    tree_estimators = (
+            ExtraTreesRegressor, RandomForestRegressor,
+            GradientBoostingQuantileRegressor)
+    return (
+        isinstance(estimator, tree_estimators) or
+        (isinstance(estimator, GaussianProcessRegressor) and space.is_categorical)
```

Comment:
Was just wondering: should we be checking if one of the kernels in the GP is a Hamming kernel to decide if there are gradients available or not? Don't know enough about the various kernels to know if Hamming could have gradients or not. Would also cover weird cases where someone uses a kernel without gradients but doesn't have a categorical space (if such kernels exist).

Tim's mission to reduce the number of function arguments 😆